### ISSUE TITLE
false positive when passing a string to a function

### ISSUE THREAD

**Issue by Mister-Hope** [stylelint-scss/stylelint-scss#1228]
Here is a function helper to encode svg string.

```scss
@function encode-svg($svg) {
  $encoded: "";
  $slice: 2000;
  $index: 0;
  $loops: math.ceil(math.div(string.length($svg), $slice));

  @for $i from 1 through $loops {
    $chunk: string.slice($svg, $index, $index + $slice - 1);
    $chunk: str-replace($chunk, '"', "'");
    $chunk: str-replace($chunk, "%", "%25");
    $chunk: str-replace($chunk, "#", "%23");
    $chunk: str-replace($chunk, "{", "%7B");
    $chunk: str-replace($chunk, "}", "%7D");
    $chunk: str-replace($chunk, "<", "%3C");
    $chunk: str-replace($chunk, ">", "%3E");
    $encoded: #{$encoded}#{$chunk};
    $index: $index + $slice;
  }

  @return url("data:image/svg+xml;utf8,#{$encoded}");
}
```

Generally speaking, the `$svg` is expected to be filled with quote strings for sure.

<img width="1378" height="157" alt="Image" src="https://github.com/user-attachments/assets/764879b4-da44-4842-a227-60f0df8672a9" />

This is absolutely a false positive, as it makes no sense to use a function like this. The correct way should be

```scss
:root {
  --a: #{svg.encode-svg($icon)};
}
```

The plugin should suggest interpolation the whole function, not the param. Since it has an "auto fix", this could be pretty annoying, as a wrong fix is applied instead.
